### PR TITLE
Make status code check more strict for sitemap plugin

### PR DESCRIPTION
**Pull request by adrianlyjak**
This is a potential fix for #10778

## Changes

- Makes status code exclusion from sitemap more strict. Only `[locale]/[error-code]` and `[error-code]` style pages are excluded
- requires locales to be configured for locale page exclusion

## Testing

Updated existing test fixture and added test for this scenario

## Docs

The current exclusion logic isn't documented, so it seems unnecessary to add further documentation about this edge case. IMO, this seems like a bug

### CODE REVIEW

**Comment by matthewp** [comment]
Can you rebase against `main`?

**Comment by adrianlyjak** [comment]
@matthewp 👌 rebased and all checks pass

**Review by bholmesdev — APPROVED** [review]
This fix looks good to me! Confirmed that 404 and 500 should only be detected from `src/pages` in our docs, and should include localized version as well per the i18n middleware. We Adri-can merge I think @adrianlyjak

File: .changeset/nasty-turtles-obey.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@astrojs/sitemap": patch
+---
+
+Site map status code exclusion now only excludes pages like /[error-code] or /[locale]/[error-code]
```
Comment (bholmesdev) [inline .changeset/nasty-turtles-obey.md:5]:
```suggestion
Fixed false positives for status code routes like `404` and `500` when generating sitemaps.
```

nit: I'd restructure to describe the bug solved a bit more clearly
